### PR TITLE
Implement Restonomer transformation - ReplaceStringInColumnValue

### PR DESCRIPTION
**Pull request by GM2222**
Implement Restonomer transformation - ReplaceStringInColumnValue

### CODE REVIEW

File: restonomer-core/src/it/resources/mock_data/transformation/__files/mocked_replace_StringValue_transformation_response.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "col_A": "val_A",
+  "col_B": "val_B",
+  "col_C": "val_C"
+}
```
Comment (rahulbhatia023) [inline restonomer-core/src/it/resources/mock_data/transformation/__files/mocked_replace_StringValue_transformation_response.json:5]: Add new line at the end of the file

File: restonomer-core/src/it/resources/mock_data/transformation/expected_replace_StringValue_transformation.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "col_A": "val_A",
+  "col_B": "value_B",
+  "col_C": "val_C"
+}
```
Comment (rahulbhatia023) [inline restonomer-core/src/it/resources/mock_data/transformation/expected_replace_StringValue_transformation.json:5]: Add new line at the end of the file

File: restonomer-core/src/it/resources/restonomer_context/checkpoints/transformation/checkpoint_replace_StringValue_transformation.conf [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      }
+    ]
+
+    persistence = {
+      type = "file-system"
+      file-format = "json"
+      file-path = "/tmp/transformation"
+    }
+  }
+}
```
Comment (rahulbhatia023) [inline restonomer-core/src/it/resources/restonomer_context/checkpoints/transformation/checkpoint_replace_StringValue_transformation.conf:27]: Add new line at the end of the file

File: restonomer-docs/source/restonomer_transformations.md [hunk 4]
```diff
@@ -282,3 +282,56 @@ The transformed response will now have the columns with the desired data types:
   "col_B": "[{'ZipCodeType':'STANDARD','Zipcode':704}]"
 }
 ```
+## ReplaceStringInColumnValue
+
+It lets the user cast the data type of multiple columns to the desired different types at once.
```
Comment (rahulbhatia023) [inline restonomer-docs/source/restonomer_transformations.md:287]: This description is wrong. Its copied from CastColumns. Please correct it.

File: restonomer-docs/source/restonomer_transformations.md [hunk 5]
```diff
@@ -282,3 +282,56 @@ The transformed response will now have the columns with the desired data types:
...
+
+This transformation expects user to provide below inputs:
+
+| Input Arguments | Mandatory | Default Value | Description                          |
+|:----------------|:---------:|:-------------:|:-------------------------------------|
+| column-Name     |    Yes    |       -       | It is the column name                |
```
Comment (rahulbhatia023) [inline restonomer-docs/source/restonomer_transformations.md:293]: Make N to small letter. It should be "column-name"

File: restonomer-docs/source/restonomer_transformations.md [hunk 6]
```diff
@@ -282,3 +282,56 @@ The transformed response will now have the columns with the desired data types:
...
+Then, user can configure the `replaceStringInColumnValue` transformation in the below manner:
+
+```hocon
+{
+  type = "replace-string-in-column-value"
+  column-name = "col_A"
+  pattern = 5
+  replacement = "abc"
+  }
+}
```
Comment (rahulbhatia023) [inline restonomer-docs/source/restonomer_transformations.md:326]: There is one extra closing bracket

File: restonomer-docs/source/restonomer_transformations.md [hunk 7]
```diff
@@ -282,3 +282,56 @@ The transformed response will now have the columns with the desired data types:
...
+```
+
+The transformed response will now have the columns with the desired data types:
```
Comment (rahulbhatia023) [inline restonomer-docs/source/restonomer_transformations.md:329]: We are not focusing on changing data types here. Pls fix this statement.

**Review by rahulbhatia023 — CHANGES_REQUESTED** [review]
Nice job. Just few comments.

File: restonomer-core/src/it/resources/mock_data/transformation/expected_replace_StringValue_transformation.json (hunk 2 above)
Comment (GM2222) [reply restonomer-core/src/it/resources/mock_data/transformation/expected_replace_StringValue_transformation.json]: Done

File: restonomer-core/src/it/resources/restonomer_context/checkpoints/transformation/checkpoint_replace_StringValue_transformation.conf (hunk 3 above)
Comment (GM2222) [reply restonomer-core/src/it/resources/restonomer_context/checkpoints/transformation/checkpoint_replace_StringValue_transformation.conf]: Done

File: restonomer-docs/source/restonomer_transformations.md (hunk 4 above)
Comment (GM2222) [reply restonomer-docs/source/restonomer_transformations.md]: Done

File: restonomer-docs/source/restonomer_transformations.md (hunk 5 above)
Comment (GM2222) [reply restonomer-docs/source/restonomer_transformations.md]: Done

File: restonomer-docs/source/restonomer_transformations.md (hunk 6 above)
Comment (GM2222) [reply restonomer-docs/source/restonomer_transformations.md]: Done

File: restonomer-docs/source/restonomer_transformations.md (hunk 7 above)
Comment (GM2222) [reply restonomer-docs/source/restonomer_transformations.md]: Done

**Review by GM2222 — COMMENTED** [review]
Hi Rahul,

I have made the changes. Can you please review it again. Thanks!!

**Review by rahulbhatia023 — APPROVED** [review]

**Pull request merged**